openupgrade_scripts/scripts/account/14.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,379 @@
+            ),
+            (
+                "res.company",
+                "res_company",
+                "accrual_default_journal_id",
+                "automatic_entry_default_journal_id",
+            ),
+        ],
+    )
+
+
+def m2m_tables_account_journal_renamed(env):
+    openupgrade.rename_tables(
+        env.cr,
+        [
+            ("account_account_type_rel", "journal_account_control_rel"),
+            ("account_journal_type_rel", "journal_account_type_control_rel"),
+        ],
+    )
+
+
+def remove_constrains_reconcile_models(env):
+    openupgrade.remove_tables_fks(
+        env.cr,
+        [
+            "account_reconcile_model_analytic_tag_rel",
+            "account_reconcile_model_account_tax_rel",
+            "account_reconcile_model_template_account_tax_template_rel",
+            "account_reconcile_model_tmpl_account_tax_bis_rel",
+        ],
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_reconcile_model_analytic_tag_rel
+        ALTER COLUMN account_reconcile_model_id DROP NOT NULL
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_reconcile_model_analytic_tag_rel
+        ADD COLUMN account_reconcile_model_line_id integer
+        """,
+    )
+
+
+def copy_fields(env):
+    openupgrade.rename_columns(
+        env.cr,
+        {
+            "account_journal": [
+                ("default_credit_account_id", None),
+                ("default_debit_account_id", None),
+            ],
+        },
+    )
+    if openupgrade.column_exists(env.cr, "account_move", "move_type"):
+        # see account_tax_balance of OCA
+        openupgrade.rename_columns(
+            env.cr,
+            {
+                "account_move": [
+                    ("move_type", None),
+                ],
+            },
+        )
+
+
+def add_move_id_field_account_bank_statement_line(env):
+    if not openupgrade.column_exists(
+        env.cr, "account_bank_statement_line", "currency_id"
+    ):
+        openupgrade.logged_query(
+            env.cr,
+            """
+            ALTER TABLE account_bank_statement_line
+            ADD COLUMN currency_id integer
+            """,
+        )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE account_bank_statement_line absl
+        SET currency_id = COALESCE(aj.currency_id, rc.currency_id)
+        FROM account_bank_statement bs
+        JOIN account_journal aj ON bs.journal_id = aj.id
+        JOIN res_company rc ON aj.company_id = rc.id
+        WHERE absl.statement_id = bs.id
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_bank_statement_line
+        ADD COLUMN move_id integer
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE account_move
+        ADD COLUMN statement_line_id integer
```

Comment:
getting an error "column statement_line_id exists in relation 'account_move', checking if it's data-specific or a recurring problem. will follow up in 30m
A check like done above may be in order.

Reply:
Is it this one?

Reply:
when older versions sneakily strike haha solved :)